modyn/config/schema/pipeline/trigger/drift/alibi_detect.py

```diff
@@ -0,0 +1,57 @@
+from typing import Annotated, Literal, Union
+
+from modyn.config.schema.base_model import ModynBaseModel
+from pydantic import Field
+
+
+class _AlibiDetectBaseDriftMetric(ModynBaseModel):
+    p_val: float = Field(0.05, description="The p-value threshold for the drift detection.")
+
+    # to be added in the future:
+
+    # x_ref_preprocessed
+    # preprocess_at_init
+    # configure_kernel_from_x_ref
+
+
+class AlibiDetectDeviceMixin(ModynBaseModel):
+    device: str | None = Field(
+        None,
+        description="The device used for the drift detection.",
+        pattern="^(cpu|gpu|cuda:?[0-9]*)$",
+    )
+
+
+class _AlibiDetectCorrectionMixin(ModynBaseModel):
+    correction: Literal["bonferroni", "fdr"] = Field("bonferroni")
+
+
+class _AlibiDetectAlternativeMixin(ModynBaseModel):
+    alternative_hypothesis: Literal["two-sided", "less", "greater"] = Field("two-sided")
+
+
+class AlibiDetectMmdDriftMetric(_AlibiDetectBaseDriftMetric, AlibiDetectDeviceMixin):
```

Comment:
How similar are Alibi Detect and Evidently? In a perfect world, instead of having  `AlibiDetect[XXX]Metric` and  `EvidentlyXXXMetric`, we would have an XX metric and a field where we either say alibi detect or evidently, i.e., it's a configuration option of the metric rather than a different metric. Do you think this is feasible?

Reply:
They support different features like hardware acceleration (incl. device selection),  kernel setup, thresholds and for most the configuration of the hypothesis tests (bootstrap vs. permutation: alibi detect allows to specify the num permutations, while evidently hard codes that). Also the quantile_probability sometimes seems to have an inverted definition (even within the same library, e.g. 0.95 instead of 0.05).

Overall I'd rather keep the abstractions close to their underlying variants.

Reply:
Yeah, makes sense. Then let's keep it how you proposed it.